pyuvdata/uvflag/uvflag.py

```diff
@@ -790,7 +886,6 @@ def _set_type_baseline(self):
         self._Nants_data.required = True
         self._Nbls.required = True
         self._Nblts.required = True
-        self._Nspws.required = True
```

Comment:
It will always be required in the future because the `spw_array` will always be required in the future. I took these out because I also removed the line in `_set_type_waterfall` that set it to False. But that might not have been the right thing to do. I'm going to poke at it a little more.

Reply:
So the reason it was required for antenna and baseline types in the past was because they had an array with a size given by Nspws. But now they don't (it's either simply size 1 or that axis doesn't exist). It will be required in future for all objects, not just antenna and baseline objects, because the `spw_array` will be required, which is size `Nspws`. But I agree that it's a little strange to be relaxing this requirement for antenna and baseline objects when it's going to be enforced again soon. I'll put these lines back.